cleanlab/internal/neighbor/neighbor.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+from typing import Optional, TYPE_CHECKING
+
+import numpy as np
+from scipy.sparse import csr_matrix
+from sklearn.neighbors import NearestNeighbors
+
+if TYPE_CHECKING:
+    from cleanlab.internal.neighbor.types import FeatureArray, Metric
+
+from cleanlab.internal.neighbor.metric import decide_metric
+from cleanlab.internal.neighbor.search import construct_knn
+
+
+DEFAULT_K = 10
```

Comment:
As the PR is only aimed at refactoring outlier-detection and near-duplicate detection. Currently, the outlier check uses DEFAULT_K as a default, while near-duplicate checks just need one neighbor (other than the query point).

Reply:
please add a comment next to this constant (same for all constants we ever introduce) explaining it.

Reply:
I thought Datalab more generally uses `DEFAULT_K` to construct its KNN graph. We are only ever constructing a single KNN graph, that will be reused across all issue-managers that are based on KNN: Outliers, Duplicates, Data Valuation, Underperforming Group, etc.

Reply:
Most types use the default value of 10, but each issue type happens to set the same default value.

In any case, I've added a comment.